kmath-core/src/commonMain/kotlin/scientifik/kmath/structures/CreationRoutines.kt

```diff
@@ -0,0 +1,169 @@
+package scientifik.kmath.structures
+
+import scientifik.kmath.operations.RealField
+import scientifik.kmath.operations.RealField.power
+import kotlin.math.*
+
+
+object RealFactories{
+    /**
+     *  Create a NDArray filled with ones
+     */
+    fun ones(vararg shape: Int) = NDElement.real(shape){1.0}
+
+    /**
+     * Create a 2D NDArray, with ones on the diagonal and zeros elsewhere.
+     *
+     * @param offset Index of the diagonal: 0 (the default) refers to the main diagonal, a positive value refers to an upper diagonal, and a negative value to a lower diagonal.
+     */
+    fun eye(dim1: Int, dim2: Int, offset : Int = 0) = NDElement.real2D(dim1, dim2){i, j -> if (i == j + offset) 1.0 else 0.0}
+
+    /**
+     * An array with ones at and below the given diagonal and zeros elsewhere.
+     * T[i,j] == 1 for i <= j + offset
+     *
+     * @param offset Index of the diagonal: 0 (the default) refers to the main diagonal, a positive value refers to an upper diagonal, and a negative value to a lower diagonal.
+     */
+    fun triangle(dim1: Int, dim2: Int, offset : Int = 0) = NDElement.real2D(dim1, dim2){i, j -> if (i <= j + offset) 1.0 else 0.0}
```

Comment:
Switch IDEA to modern codestyle convention: Settings->Editor->Code Style->Kotlin->set from->predefined-> Kotlin style guide